modules/integration/tests-common/common-artifacts/lazy-loading-info/src/main/java/org/wso2/appserver/integration/common/artifacts/lazyloading/LazyLoadingInfoUtil.java

```diff
@@ -0,0 +1,164 @@
+            log.info("Tenant " + tenantDomain + " loaded :" + isTenantContextLoaded);
+        }
+        return new TenantStatus(isTenantContextLoaded);
+    }
+
+    /**
+     * Check  the given web-app of given tenant is loaded.
+     *
+     * @param tenantDomain Tenant name
+     * @param webAppName   Web-app Name
+     * @return WebAppStatus  with current status  information about the Web app.
+     */
+    protected static WebAppStatus getWebAppStatus(String tenantDomain, String webAppName) {
+        WebAppStatus webAppStatus = new WebAppStatus();
+        ConfigurationContext tenantConfigurationServerContext = getTenantConfigurationServerContext(tenantDomain);
+        if (tenantConfigurationServerContext != null) {
+            webAppStatus.setTenantStatus(new TenantStatus(true));
+            log.info("Tenant " + tenantDomain + " configuration context is loaded.");
+            WebApplicationsHolder webApplicationsHolder = (WebApplicationsHolder) ((HashMap)
+                    tenantConfigurationServerContext.getLocalProperty(CARBON_WEBAPPS_HOLDER_LIST)).get(WEBAPPS);
+            Map<String, WebApplication> startedWebAppMap = webApplicationsHolder.getStartedWebapps();
+            if (startedWebAppMap != null) {
+                WebApplication webApplication = startedWebAppMap.get(webAppName);
+                if (webApplication != null) {
+                    webAppStatus.setWebAppStarted(true);
+                    log.info("Tenant " + tenantDomain + " Web-app: " + webAppName + " is available in configuration context.");
+                    boolean isWebAppGhost = Boolean.parseBoolean((String) webApplication.getProperty(GHOST_WEB_APP));
+                    log.info("Tenant " + tenantDomain + " Web-app: " + webAppName + " is in Ghost deployment status :" +
+                            isWebAppGhost);
+                    webAppStatus.setWebAppGhost(isWebAppGhost);
+                } else {
+                    log.info("Given web-app:" + webAppName + " for tenant:" + tenantDomain + " not found in started state");
+                    webAppStatus.setWebAppStarted(false);
+                }
+            } else {
+                log.info("Tenant " + tenantDomain + " has no started web-apps.");
+                webAppStatus.setWebAppStarted(false);
+            }
+        } else {
+            log.info("Tenant " + tenantDomain + " configuration context is not loaded.");
+            webAppStatus.setTenantStatus(new TenantStatus(false));
+        }
+        return webAppStatus;
+    }
+
+
+    /**
+     * Check  the given web-app of super tenant is loaded.
+     *
+     * @param webAppName Web-app Name
+     * @return WebAppStatus  with current status  information about the Web app.
+     */
+
+    protected static WebAppStatus getSuperTenantWebAppStatus(String webAppName) {
+        WebAppStatus webAppStatus = new WebAppStatus();
+        webAppStatus.setTenantStatus(new TenantStatus(true)); // Super tenant always loaded;
```

Comment:
@prabathabey  I have used bean class to keep the status of the tenant  rather using boolean because of, when we return the value from the REST service it will give more meaning full JSON object.